src/LazyEventDispatcher.php

```diff
@@ -64,4 +64,26 @@ public function addSubscriberLazy(string $eventName, string $serviceName): void
 		$this->mapping[$eventName][] = $serviceName;
 	}
 
+	/**
+	 * @param string|null $eventName
+	 * @phpcsSuppress SlevomatCodingStandard.TypeHints.TypeHintDeclaration.MissingParameterTypeHint
+	 */
+	public function hasListeners($eventName = null): bool
+	{
+		// check if event name is specified and has some lazy subscriber
+		if ($eventName !== null && isset($this->mapping[$eventName])) {
+			return true;
+		}
+
+		// check if any lazy subscriber exists
+		if ($eventName === null) {
+			foreach ($this->mapping as $serviceName) {
```

Comment:
What about `return !empty($this->mapping)`?

Reply:
Right, that would work too I think.
I was too locked on the way the original implementation is written :-)